information/specification.md

```diff
@@ -0,0 +1,830 @@
+outputs from a wallet's [UTxO set](#utxo-set) in order to make one or more
+payments, and computing the set of change to be paid back to the wallet.
+
+## Why is Coin Selection Non-Trivial?
+
+### Issues
+
+There are a number of issues which make the problem of coin selection more
+complicated than it would initially appear.
+
+ * Each [transaction](#transaction) has a _maximum size_, as defined by the
+   protocol. The size of a transaction increases as we add more
+   [inputs](#transaction-input) or [outputs](#transaction-output).
+
+   Therefore, there's a practical limit on the number of coins we can select
+   for any given transaction.
+
+ * The most obvious strategy for coin selection, which consists of trying to
+   get as close to the requested value as possible, will tend (over time) to
+   create a lot of [dust outputs](#dust-output), outputs with small associated
+   values that are not worth including in a transaction.
+
+   Dust outputs are a problem, because even if the total value of dust in a
+   wallet is more than enough to cover a given target amount, if we attempt to
+   include that dust in a given transaction, we may run out of space (by
+   reaching the transaction size limit) before we can cover the target amount.
+
+ * The most obvious strategy for paying change, which consists of making a
+   single [change output](#change-output) with the exact excess value, will
+   tend (over time) to reduce the size of a [UTxO set](#utxo-set). This is bad
+   for two reasons:
+
+    1. Having a small UTxO set limits the number of payments that we can make
+       in parallel.
+
+    2. The approach of coalescing all change into a single output is widely
+       considered to have negative privacy implications. The discussion of
+       privacy issues is beyond the scope of this article.
+
+### Goals
+
+There are several desirable properties that we would like a coin selection
+algorithm to have.
+
+In particular, a coin selection algorithm should:
+
+  * employ strategies to limit the amount of [dust](#dust-output) that
+    accumulates in the [UTxO set](#utxo-set).
+
+  * over the course of time, aim to generate and maintain a [UTxO
+    set](#utxo-set) with _useful_ outputs: that is, outputs that allow us to
+    process future payments with a reasonably small number of
+    [inputs](#transaction-input).
+
+For more information on dust avoidance, see [Self Organisation in Coin
+Selection](#self-organisation-in-coin-selection).
+
+# Definitions
+
+### Address
+
+An _address_ is a unique identifier that represents a destination for a payment.
+
+Addresses are typically owned (and generated) by individual wallets.
+
+In general, coin selection algorithms are agnostic to the choice of address
+format. Therefore, the details of individual address formats are not described
+within this document.
+
+### Coin Value
+
+A _coin value_ is a positive integer value that represents a number of
```

Comment:
This should be changed back to **_non-negative_**.

See [this comment](https://github.com/input-output-hk/cardano-coin-selection/issues/38#issuecomment-618262828) on issue https://github.com/input-output-hk/cardano-coin-selection/issues/38.